adapters/resetdigital/resetdigital.go

```diff
@@ -0,0 +1,301 @@
+			tempH = *imp.Banner.H
+		}
+		if imp.Banner.W != nil {
+			tempW = *imp.Banner.W
+		}
+		if tempH > 0 && tempW > 0 {
+			reqData.Imps[0].MediaTypes.Banner.Sizes = append(
+				reqData.Imps[0].MediaTypes.Banner.Sizes,
+				[]int64{tempW, tempH},
+			)
+		}
+	}
+	if bidType == openrtb_ext.BidTypeVideo && imp.Video != nil {
+		var tempH, tempW int64
+		if imp.Video.H != nil {
+			tempH = *imp.Video.H
+		}
+		if imp.Video.W != nil {
+			tempW = *imp.Video.W
+		}
+		if tempH > 0 && tempW > 0 {
+			reqData.Imps[0].MediaTypes.Video.Sizes = append(
+				reqData.Imps[0].MediaTypes.Video.Sizes,
+				[]int64{tempW, tempH},
+			)
+		}
+	}
+
+	var bidderExt adapters.ExtImpBidder
+	var resetDigitalExt openrtb_ext.ImpExtResetDigital
+
+	if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	if err := json.Unmarshal(bidderExt.Bidder, &resetDigitalExt); err != nil {
+		return resetDigitalRequest{}, err
+	}
+	reqData.Imps[0].ZoneID.PlacementID = resetDigitalExt.PlacementID
+
+	return reqData, nil
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if adapters.IsResponseStatusCodeNoContent(responseData) {
+		return nil, nil
+	}
+
+	if err := adapters.CheckResponseStatusCodeForErrors(responseData); err != nil {
+		return nil, []error{err}
+	}
+
+	var response resetDigitalBidResponse
+	if err := json.Unmarshal(responseData.Body, &response); err != nil {
+		return nil, []error{err}
+	}
+
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(len(request.Imp))
+
+	var errs []error
+	requestImps := make(map[string]openrtb2.Imp)
+	for _, imp := range request.Imp {
+		requestImps[imp.ID] = imp
+	}
+
+	for i := range response.Bids {
+		resetDigitalBid := &response.Bids[i]
+
+		bid, err := getBidFromResponse(resetDigitalBid)
+		// handle the error
+		if bid == nil {
+			// it would be better to return an error here
+			continue
+		}
+
+		bidType, err := GetMediaTypeForImp(requestImps, bid.ImpID)
+		if err != nil {
+			return nil, []error{err}
+		}
+
+		b := &adapters.TypedBid{
+			Bid:     bid,
+			BidType: bidType,
+			Seat:    openrtb_ext.BidderName(resetDigitalBid.Seat),
+		}
+		bidResponse.Bids = append(bidResponse.Bids, b)
+	}
+
+	if len(request.Cur) == 0 {
+		bidResponse.Currency = "USD"
+	}
+
+	return bidResponse, errs
+}
+
+func getBidFromResponse(bidResponse *resetDigitalBid) (*openrtb2.Bid, error) {
+	if bidResponse.CPM == 0 {
+		// brian to check how to report this
+		return nil, nil
+	}
+
+	bid := &openrtb2.Bid{
+		ID:    bidResponse.BidID,
+		Price: bidResponse.CPM,
+		ImpID: bidResponse.ImpID,
+		CID:   bidResponse.CID,
+		CrID:  bidResponse.CrID,
+		AdM:   bidResponse.HTML,
+	}
+
+	if i, err := strconv.ParseInt(bidResponse.W, 10, 64); err == nil && i > 0 {
+		bid.W = i
+	} else if err != nil {
+		return nil, err
+	} // the error should be returned here if ParseInt fails
```

Comment:
A better way to write this would be:
```
w, err := strconv.ParseInt(bidResponse.W, 10, 64)
if err != nil {
	return nil, err
}
bid.W = w
```
This matches the current behavior you have. If the width is successfully converted to an int but is zero, what do you want to do?